pathway-analysis/02-gene_set_enrichment_analysis.Rmd

```diff
@@ -111,77 +107,143 @@ dge_results_df <- readr::read_tsv(dge_results_file)
 head(dge_results_df)
 ```
 
-We can take the same steps we did earlier to convert these Ensembl gene IDs to Entrez IDs. 
-We have to change the `column` argument to `mapIds()` (and the downstream steps).
+Since this data frame of DGE results includes gene symbols, we do not need to perform any kind of gene identifier conversion.
+We do, however, need to check for duplicate gene symbols. 
+We can accomplish this with `duplicated()`, which returns a logical vector (e.g., `TRUE` or `FALSE`).
+The function `any()` will return `TRUE` if **any** value it is given `TRUE`.
+
+```{r any_duplicated, live = TRUE}
+any(duplicated(dge_results_df$gene_symbol))
+```
+
+This will cause a problem when we go to run GSEA.
+Let's talk more about the method itself to understand _why_.
+
+## Gene Set Enrichment Analysis
+
+_Adapted from [refine.bio examples](https://github.com/AlexsLemonade/refinebio-examples/blob/33cdeff66d57f9fe8ee4fcb5156aea4ac2dce07f/03-rnaseq/pathway-analysis_rnaseq_02_gsea.Rmd)_
+
+![](diagrams/subramanian_fig1.jpg)
+
+**Figure 1. [Subramanian, Tamayo *et al.* (2005)](https://doi.org/10.1073/pnas.0506580102).**
+
+GSEA calculates a pathway-level metric, called an enrichment score (sometimes abbreviated as ES), by ranking genes by a gene-level statistic.
+This score reflects whether or not a gene set or pathway is overrepresented at the top or bottom of the gene rankings ([Subramanian, Tamayo *et al.* 2005](https://doi.org/10.1073/pnas.0506580102); [Yu](http://yulab-smu.top/clusterProfiler-book/chapter2.html#gene-set-enrichment-analysis)) 
+
+Specifically, genes are ranked from most positive to most negative based on their statistic and a running sum is calculated by starting with the most highly ranked genes and increasing the score when a gene is in the pathway and decreasing the score when a gene is not.
```

Comment:
Running sum of what? Are we adding up the statistics or an indicator?

Reply:
I _believe_ in this particular implementation it will be the statistics, but will double check.

Reply:
Okay I remember why I don't get _into it_ here. I believe it's the absolute value of the ranked statistic scaled by the `sum(abs(< stats of genes in pathway>))` for the "hits" (addition) and then `1 / <number of genes outside the pathway>` for the "misses" (subtraction).  Which is not more helpful than the intuition about what the final score tells you about in my opinion?

Reply:
I rearranged the notebook (so this says outdated), but I left the sentence the way it is. Happy to take another go at it if you have suggestions!